app/models/search_engine.rb

```diff
@@ -0,0 +1,42 @@
+class SearchEngine
+  def search(datasets, search)
```

Comment:
To mi uplne nedava zmysel, kedze search engine je objekt co zije napriec requestami. datasets/search sa meni z kazdym requestom.